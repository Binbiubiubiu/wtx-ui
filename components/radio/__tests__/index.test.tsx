import React from 'react';
import { render } from 'enzyme';
import { Radio, RadioButton } from '..';
import toJson from 'enzyme-to-json';

describe('Radio', () => {
  it('should render Radio components', () => {
    const wrapper = render(
      <>
        <Radio.Group value="七天">
          <Radio label="三天" />
          <Radio label="七天" />
          <Radio label="三十天" />
        </Radio.Group>
        <RadioButton.Group value="图表">
          <RadioButton label="图表">图表</RadioButton>
          <RadioButton label="列表">列表</RadioButton>
        </RadioButton.Group>
      </>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should style output is not null', () => {
    expect(require('../style')).not.toBeNull();
  });
});
