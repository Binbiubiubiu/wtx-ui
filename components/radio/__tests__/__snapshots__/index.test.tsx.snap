// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render Radio components 1`] = `
Array [
  <div
    class="wtx-radio-group"
  >
    <label
      class="wtx-radio"
    >
      <input
        class="wtx-radio__input"
        name="wtx-radio-0"
        type="radio"
        value="三天"
      />
      三天
    </label>
    <label
      class="wtx-radio wtx-radio__checked"
    >
      <input
        class="wtx-radio__input"
        name="wtx-radio-1"
        type="radio"
        value="七天"
      />
      七天
    </label>
    <label
      class="wtx-radio"
    >
      <input
        class="wtx-radio__input"
        name="wtx-radio-2"
        type="radio"
        value="三十天"
      />
      三十天
    </label>
  </div>,
  <div
    class="wtx-radio-group"
  >
    <label
      class="wtx-radio-button wtx-radio-button__checked"
    >
      <input
        class="wtx-radio-button__input"
        name="wtx-radio-0"
        type="radio"
        value="图表"
      />
      图表
    </label>
    <label
      class="wtx-radio-button"
    >
      <input
        class="wtx-radio-button__input"
        name="wtx-radio-1"
        type="radio"
        value="列表"
      />
      列表
    </label>
  </div>,
]
`;
